backend/internal/activities/deployer_activities.go

```diff
@@ -636,3 +648,84 @@ func getConfig(state ewf.State) (statemanager.ClientConfig, error) {
 
 	return config, nil
 }
+
+func FetchKubeconfigStep(privateKeyPath string) ewf.StepFn {
+	return func(ctx context.Context, state ewf.State) error {
+		cluster, err := statemanager.GetCluster(state)
+		if err != nil {
+			return fmt.Errorf("failed to get cluster from state: %w", err)
+		}
+
+		var master *kubedeployer.Node
+		for i := range cluster.Nodes {
+			if cluster.Nodes[i].Type == kubedeployer.NodeTypeLeader || cluster.Nodes[i].Type == kubedeployer.NodeTypeMaster {
+				master = &cluster.Nodes[i]
+				break
+			}
+		}
+		if master == nil {
+			return fmt.Errorf("no leader or master node found in cluster")
+		}
+
+		privateKeyBytes, err := os.ReadFile(privateKeyPath)
+		if err != nil {
+			return fmt.Errorf("failed to read SSH private key: %w", err)
+		}
+
+		kubeconfig, err := internal.GetKubeconfigViaSSH(string(privateKeyBytes), master)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve kubeconfig via SSH: %w", err)
+		}
+
+		state["kubeconfig"] = kubeconfig
+		return nil
+	}
+}
+
+func VerifyClusterReadyStep() ewf.StepFn {
```

Comment:
we need to update the notification message, to increment the total number of steps by 1, so it should be `nodesCount + 2`, for network step + verification step